### ISSUE TITLE
Fix and refactor npm build process

### ISSUE THREAD

**Issue by WGierke** [concept-to-clinic/concept-to-clinic#106]
Currently, when attempting to build the new npm front-end, I'm running into a few issues. Firstly, it's not enough anymore to run `docker-compose -f local.yml build && docker-compose -f local.yml up` like it's said by the documentation. Now, one also needs to run `npm install` and `npm run build` manually in the `frontend` directory. Without these commands, the front-end could not be served. Unfortunately, the latter command throws an error while building (which is easy to fix). However, even after fixing it I can only see the navigation but no content.

## Expected Behavior
The documentation says that building the docker containers and starting them should be enough to run the application locally. One would also expect to see the content we already implemented until now.

## Current Behavior
One also has to run some npm commands manually, which throws an error. Even after fixing that, only the navbar can be seen but no content like we had it before with the Django templates.

## Possible Solution
- move the npm commands in the Dockerfiles (might be cleaner than adding them to the documentation)  
- adapt the image path that's throwing an error during `npm run build`

## Steps to Reproduce
Check out the current master  
`git reset --hard origin/master`
Remove all files and folders that are not explicitly tracked by Git (such as npm build files or modules)  
`sudo git clean -x -i -d`
Build and start the docker containers
`sudo docker-compose -f local.yml build && sudo docker-compose -f local.yml up`  
When visiting http://localhost:8000/, you'll get a `TemplateNotFoundError`.
![screenshot from 2017-09-10 18-54-20](https://user-images.githubusercontent.com/6676439/30251086-7ee554e6-9659-11e7-8000-e52a6df8b318.png)
Thus, we need to install the npm dependencies manually and compile the application  
`cd interface/frontend && npm install && npm run build`
You should get the error `Module not found: Error: Can't resolve '../assets/c-logo-light.svg' in '/app/frontend/src/components'`.
![screenshot from 2017-09-10 18-56-17](https://user-images.githubusercontent.com/6676439/30251102-c5117e72-9659-11e7-8a7c-5e85cceb6234.png)
This can be solved by adapting the image path (only a hot fix):
```
--- a/interface/frontend/src/components/AppNav.vue
+++ b/interface/frontend/src/components/AppNav.vue
@@ -1,6 +1,6 @@
 <template>
 <nav class="navbar navbar-toggleable-md navbar-inverse bg-inverse fixed-top">
-  <img src="../assets/c-logo-light.svg"/>
+  <img src="../assets/images/c-logo-light.svg"/>
   <a class="navbar-brand" href="#">Concept To Clinic</a>
```
`npm run build` should now succeed.  
However, when visiting http://localhost:8000/, I can only see the navbar without any content when switching tabs:
![screenshot from 2017-09-10 18-48-10](https://user-images.githubusercontent.com/6676439/30251040-b4b7fffc-9658-11e7-9074-89e216896ef6.png)

I'd highly appreciate any feedback about that. Maybe I f*cked up something really hard during building the new front-end?

## Checklist before submitting
- [X] I have confirmed this using the officially supported Docker Compose setup using the `local.py` configuration and ensured that I built the containers again and they reflect the most recent version of the project at the `HEAD` commit on the `master` branch 
- [X] I have searched through the other currently open issues and am confident this is not a duplicate of an existing bug
- [X] I provided a **minimal code snippet** or list of steps that reproduces the bug.
- [X] I provided **screenshots** where appropriate
- [X] I filled out all the relevant sections of this template

**Comment by antonow** [comment]
@WGierke, the issue with no content appearing is not your fault. When I started working on the frontend build process, I made a recommendation to serve all pages (other than the initial application load) through Vue, which @isms seemed to agree with [here](https://github.com/concept-to-clinic/concept-to-clinic/issues/6#issuecomment-325644418). In the meantime, you were completing issues using a different scheme in which the base views for each page were rendered from Django. These pages and content that you were working on will need to be ported over to the SPA scheme that I set up in [PR 97](https://github.com/concept-to-clinic/concept-to-clinic/pull/97) or, if the project supervisors think that we should use the scheme that you have been developing with, then we will need to update the frontend build process.

**Comment by WGierke** [comment]
@antonow Thanks for the clarification! Sorry, I did not notice that conversation. 
So it might make sense to file a new issue about porting the front-end we already had with Django templates to Vue (@isms @lamby @reubano )? Otherwise, the implementation of the remaining front-end issues #30 #33 #34 might become problematic

**Comment by isms** [comment]
@WGierke @antonow The pages should be adapted from Django templates to pages in the Vue app which interact with the backend API. The functionality should be broken down into Vue components.

I'm working on a fix now which corrects a number of problems introduced by changes to the `interface` Dockerfile (e.g. wrong Python version, 2.7 by default in Ubuntu 16.04 image.

**Comment by isms** [comment]
Closed by 7ae3c47c32c84994f977b7349221cb1d85300172. Thanks @antonow and @WGierke for looking into this.
